parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -2785,6 +2785,12 @@ opt_block_args_tail:
                         lambda->end);
                       DIAGCHECK();
                     }
+                | t_lambda error
+                    {
+                      driver.lex.unextend();
+                      driver.lex.context.inLambda = $1->context.inLambda;
+                      YYERROR;
```

Comment:
Usually what we try do in these cases is actually build a successful parse.

You can use `driver.build.error_node` to build an error node (which is just a constant literal called `<ErrorNode>` to build a successful parse, so something like, where you have the arg list that you parsed, but with an `error_node` for the lambda body.

Reply:
I've switched to explicitly building an error node in the result, instead of unconditionally calling `YYERROR`.

Reply:
I'm explicitly leaving the argument list out of the result, as it introduces shift/reduce and reduce/reduce conflicts to add a production that handles errors after the argument list. This is related to not being able to reliably call `cmdarg.pop` as without that other case it's not obvious how we can tell if we've seen the call to `cmdarg.push(false)` or not.